src/Providers/Http/HttpTransporter.php

```diff
@@ -91,6 +101,122 @@ public function send(Request $request): Response
         return $this->convertFromPsr7Response($psr7Response);
     }
 
+    /**
+     * Determines if the underlying client matches the Guzzle client shape.
+     *
+     * @since n.e.x.t
+     *
+     * @param ClientInterface $client The HTTP client instance.
+     * @return bool True when the client exposes Guzzle's send signature.
+     */
+    private function isGuzzleClient(ClientInterface $client): bool
```

Comment:
Love this! It's a really creative and pragmatic solution which, while hacky, is exactly what we need to make for a good DX for most, without forcing Guzzle on everyone. 🙌 

And with these methods being private, there's no risk. If we think this is bad later, we could simply remove it.